server/src/main/java/org/elasticsearch/action/search/SearchRequest.java

```diff
@@ -251,6 +251,7 @@ private SearchRequest(
         this.waitForCheckpointsTimeout = searchRequest.waitForCheckpointsTimeout;
         this.forceSyntheticSource = searchRequest.forceSyntheticSource;
         this.projectRouting = searchRequest.projectRouting;
+        this.resolvedIndexExpressions = searchRequest.resolvedIndexExpressions;
```

Comment:
I'd add a quick unit test for this ctor just to ensure when we refactor it next time this aspect is not lost.